lib/helpers-linter.js

```diff
@@ -0,0 +1,175 @@
+/* @flow */
+
+// Note: the following code is based on
+// https://github.com/facebook/flow/blob/v0.47.0/tsrc/flowResult.js
+// and adjusted to output nicely formatted markdown.
+
+import { format } from 'util'
+import type { StatusError, StatusExtra, StatusMessage, Location } from './types'
+
+function fileUrl({ source, start }: Location): string {
+  // build a link that can be opened by the linter ui.
+  // e.g. atom://linter?file=<path>&row=<number>&column=<number>
+  return 'atom://linter?file=' + source +
+    (start.line > 0
```

Comment:
Instead of doing it inline can we do it in a variable above this line for verbosity?